tests/test_rpm_verifier.py

```diff
@@ -355,24 +356,34 @@ def mock_image_processor(self, monkeypatch: MonkeyPatch) -> MagicMock:
         return mock
 
     @pytest.fixture()
-    def mock_generate_output(self, monkeypatch: MonkeyPatch) -> MagicMock:
-        """Monkey-patched generate_output"""
-        mock = create_autospec(generate_output, return_value=(False, ""))
-        monkeypatch.setattr(rpm_verifier, generate_output.__name__, mock)
-        return mock
+    def create_generate_output_mock(
+        self, monkeypatch: MonkeyPatch
+    ) -> Callable[[bool], MagicMock]:
+        """Create a generate_output mock with different results according to
+        the `fail_unsigned` flag"""
+
+        def _mock_generate_output(fail_unsigned: bool = False) -> MagicMock:
```

Comment:
```suggestion
        def _mock_generate_output(fail: bool = False) -> MagicMock:
```

With `fail_unsigned` we say whether we should fail if there are unsigned rpms. In this case, the input here should just tell us whether we fail or not - regardless of whether there were unsigned rpms.

I think this is also causing unclarity with the inputs we provide later to this function.
At one time we provide it with a variable called `fail_on_unsigned_rpms`, and on another we call it with `unsigned_rpm_exist`. Those are both incorrect as far as I see it. Instead we should call it with a variable called `fail` or `should_fail`.

Reply:
I changed this one to fail, and aligned the variable on the tests to `fail_on_unsigned_rpms`. I think it is important for readability sake.